packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -0,0 +1,95 @@
+/** Handles the lifecycle of several pouches */
+
+import PouchDB from 'pouchdb'
+import fromPairs from 'lodash/fromPairs'
+import map from 'lodash/map'
+import repeatWithDelay from './repeat'
+
+const DEFAULT_DELAY = 30 * 1000
+
+/* Create a cancellable promise for replication with default options */
+const startReplication = (pouch, url) => {
+  let replication
+  const promise = new Promise((resolve, reject) => {
+    replication = pouch.replicate.from(url, {
+      batch_size: 1000 // we have mostly small documents
+    })
+    replication.on('error', reject).on('complete', resolve)
+  })
+
+  const cancel = () => {
+    if (replication) {
+      replication.cancel()
+    }
+  }
+
+  promise.cancel = cancel
+  return promise
+}
+
+/**
+ * - Creates/Destroys the pouches
+ * - Replicates periodically
+ */
+export default class PouchManager {
+  constructor(doctypes, options) {
+    this.pouches = fromPairs(
+      doctypes.map(doctype => [doctype, new PouchDB(doctype)])
+    )
```

Comment:
`_.fromPairs` could be replaced by `Array.prototype.reduce` (and the third-party library will not be required anymore):

```js
doctypes.reduce((pouches, doctype) => ({ ...pouches, [doctype]: new PouchDB(doctype) }), {})
```

See a [demo](https://codesandbox.io/s/5xwqxl7xkl)